recipes/wt/all/conanfile.py

```diff
@@ -153,7 +153,7 @@ def _configure_cmake(self):
 
         def _gather_libs(p):
             libs = self.deps_cpp_info[p].libs + self.deps_cpp_info[p].system_libs
-            if not getattr(self.options[p], "shared", False):
+            if "shared" not in self.options[p] or not self.options[p].shared:
```

Comment:
Seems similar, what is broken in previous code?

Reply:
the fallback value is never used. instead an exception is raised if the option does not exist : https://github.com/conan-io/conan/pull/7303/files#diff-995bfcf42a70f07f1bb99ea604c890bbc04ca805464d4969cec3f17168c91855R75
We are just lucky here that all the dependencies actually have a shared option (https://github.com/conan-community/conan-opencv/pull/66 was not so lucky)

Reply:
By the way, this condition is fragile, because you can't be sure that all libs of a package are shared when shared option is True.

Reply:
What do you suggest ?

Reply:
to remove this condition, always propagate transitive dependencies, since recipes lack public/private dependencies information and granularity of shared/static components.